main/src/main/java/cgeo/geocaching/connector/gc/ImportBookmarkLinks.java

```diff
@@ -9,13 +9,90 @@
 import android.net.Uri;
 import android.os.Bundle;
 
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 
 import java.util.regex.Pattern;
 
+import javax.annotation.RegEx;
+
 public class ImportBookmarkLinks extends AppCompatActivity {
 
+    private final UrlToIdParser intentUrlParser = defaultBookmarkListUrlToIdParser();
+
+    public static final String BOOKMARK_LIST_API_PREFIX =
+            "https://www.geocaching.com/plan/api/gpx/list/";
+
+    interface UrlToIdParser {
+        @Nullable String tryExtractFromIntentUrl(@Nullable String intentUrl);
+    }
+
+    static class SingleUrlToIdParser implements UrlToIdParser {
+        private final Pattern matcherPattern;
+        // Android API 21 apparently does not support named capturing groups in Pattern
```

Comment:
Thats correct (and very sad), unfortunately we can't use group capturing in c:geo....

What I usually do however is put the group names into the regex anyway (for better readability of the pattern) and comment somewhere which group index corresponds to which group. If at some day we drop support for SDK21 we will be able to use named capturing group.

Reply:
Done